### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Functor.ReflectsIsomorphisms

### PR DESCRIPTION
**Pull request by mattrobball**
---

- [x] depends on: #2331

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2331
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2331~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: scripts/start_port-macos.sh [hunk 1]
```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
```
Comment (jcommelin) [inline scripts/start_port-macos.sh:1]: Idem

Comment (mattrobball) [reply scripts/start_port-macos.sh:1]: 😭

**Comment by jcommelin** [comment]
LGTM

bors d+

**Comment by mattrobball** [comment]
bors merge

**Pull request closed without merging**